fix(docs): clarified usage of validation with fine grained responses (#4101)
awslabs/aws-lambda-powertools-python: docs/core/event_handler/api_gateway.md

    --8<-- "examples/event_handler_rest/src/fine_grained_responses_output.json"
    ```

???- note "Using `Response` with data validation?"

>     When using the [data validation](#data-validation) feature with `enable_validation=True`, you need to specify the concreate type for the `Response` class. This is necessary for the validation middleware to infer the underlying type and perform the validation correctly.

Reviewer: ```suggestion
    When using the [data validation](#data-validation) feature with `enable_validation=True`, you must specify the concrete type for the `Response` class. This allows the validation middleware to infer the underlying type and perform validation correctly.
```
